build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoPublish.java

```diff
@@ -54,6 +53,8 @@
      */
     public GoPublish(ArtifactoryBuildInfoClientBuilder clientBuilder, ArrayListMultimap<String, String> properties, String repo, Path path, String version, Log logger) throws  IOException {
         super(clientBuilder, path, logger);
+        this.goDriver = new GoDriver(GO_CLIENT_CMD, null, path.toFile(), logger);
+        this.moduleName = goDriver.getModuleName();
```

Comment:
These 2 lines repeat in GoRun. Please move them up to GoCommand.

Reply:
I don't see the benefit of doing that.
In each case it happens at different stage,  and moving it to a dedicated method seems more confusing than helping.

Reply:
Can we do both in the constructor of GoCommand?